sql/parser/builtins.go

```diff
@@ -66,9 +67,16 @@ func (b builtin) match(types typeList) bool {
 // sorted please.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin1(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
-	}, DummyInt)},
+	"length": {
+		stringBuiltin1(func(s string) (Datum, error) {
+			return DInt(utf8.RuneCountInString(s)), nil
+		}, DummyInt),
+		bytesBuiltin1(func(s string) (Datum, error) {
+			return DInt(len(s)), nil
+		}, DummyInt),
+	},
+
+	// TODO(pmattis): What string functions should also support bytesType?
```

Comment:
Nevermind, I misread `to_hex`. For the hash functions, they make sense on *bytes*, not *characters*. If we follow the go way, where utf8 is effectively the only encoding, it's fine to say that a string can be both a sequence of characters and a sequence of utf8 bytes. If we want to recognize other encodings, then we need to maintain a strict separation between byte strings and character strings.